### ISSUE TITLE
Introspection Needs an overhaul

### ISSUE THREAD

**Issue by stevan** [stevan/p5-Graph-QL#24]
This work was started in eaef95ee56fae55f1fe0cf5c19d137d948986ce0 

We still need to do away with the [`enable_for_schema`](https://github.com/stevan/p5-Graph-QL/blob/a5664e6f7df4c82c80d185a8b57c4269452d044e/lib/Graph/QL/Introspection.pm#L25) and [`enable_for_resolvers`](https://github.com/stevan/p5-Graph-QL/blob/a5664e6f7df4c82c80d185a8b57c4269452d044e/lib/Graph/QL/Introspection.pm#L49) class methods, both of them will make clones of the `Schema` and `Resolver` objects respectively and I am not a fan of that (though it is a shallow clone and since everything is immutable thats safe). 

Most all other fields are now "implicit" and are handled in a way similar to the `BuiltIn::Scalars` in that they will return constant values (technically `state` variables, so the are not actually created if they are never called, but afterwards return the same values without recalculating).  

- [`Query` introspection fields (`__schema`, `__type(name)`)](https://github.com/stevan/p5-Graph-QL/blob/a5664e6f7df4c82c80d185a8b57c4269452d044e/lib/Graph/QL/Introspection.pm#L64)
- [Resolvers for those `Query` fields](https://github.com/stevan/p5-Graph-QL/blob/master/lib/Graph/QL/Introspection.pm#L91)
- [Schema Meta Model](https://github.com/stevan/p5-Graph-QL/blob/master/lib/Graph/QL/Introspection.pm#L106)

However, a few things still need attention, like the implicit `Query` fields are currently added to all `Object` types, not just the `Query` instance, and we have no support for `__typename`. 

So basically this ticket needs the following tasks completed to be closed. 

- [ ] Handle `__typename` 
- [ ] Remove the need for `enable_for_schema` by making what it does "implicit"
- [ ] Remove the need for `enable_for_resolvers` by making what it does "implicit"
- [ ] Restrict the `__schema` and `__type(name)` fields to the `Query` class (simply Proxy class should suffice)

**Comment by stevan** [comment]
Much of this has been fixed in eaef95ee56fae55f1fe0cf5c19d137d948986ce0 

We still need to do away with the `enable_for_schema` and `enable_for_resolvers` class methods, but most all other fields are now "implicit" and are handled in a way similar to the `BuiltIn::Scalars`.  However, a few things still need attention, like the implicit fields are currently added to all `Object` types, not just the `Query` instance, and we have no support for `__typename`. 

So basically this ticket needs the following tasks completed to be closed. 

- [ ] Handle `__typename` 
- [ ] Remove the need for `enable_for_schema` by making what it does "implicit"
- [ ] Remove the need for `enable_for_resolvers` by making what it does "implicit"
- [ ] Restrict the `__schema` and `__type(name)` fields to the `Query` class (simply Proxy class should suffice)
